src/prefect/server/events/actions.py

```diff
@@ -261,7 +263,18 @@ async def events_api_client(
 
     def reason_from_response(self, response: Response) -> str:
         # TODO: handle specific status codes here
-        return f"Unexpected status from {self.type} action: {response.status_code}"
+        try:
+            error_detail = response.json().get("detail")
+        except Exception:
+            error_detail = None
+        if response.status_code in {409, 422}:
+            if error_detail:
+                return f"Validation error occurred for {self.type!r} - {error_detail}"
```

Comment:
```suggestion
            if error_detail or response.status_code == 422:
                return f"Validation error occurred for {self.type!r} - {error_detail}"
```

Maybe?  We wouldn't want the 422 case saying it was a "Conflict (409)" either